### PR TITLE
feat: add did-eth identifier to specification draft

### PR DESCRIPTION
**Pull request by nickreynolds**
- one thing to think about is should the resolved DID referenced by ENS Name be identical to the DID referenced by that ENS Name's Controller? Or should they be separate DIDs?

### CODE REVIEW

**Comment by nickreynolds** [comment]
most of this was taken from existing did-ethr spec, so credit to those original authors!

File: did-eth-method-draft.md [hunk 1]
```diff
@@ -16,15 +16,51 @@
...
-```
+A DID that uses this method MUST begin with the following prefix: `did:eth`. Per the DID specification, this string
+MUST be in lowercase. The remainder of the DID, after the prefix, is specified below.
+
+## Method Specific Identifier
+
+The method specific identifier is represented as the HEX-encoded secp256k1 public key (in compressed form),
+or the corresponding HEX-encoded Ethereum address on the target network, prefixed with `0x`.
+
+    eth-did = "did:ethr:" eth-specific-identifier
```
Comment (mirceanis) [inline did-eth-method-draft.md:31]:
```suggestion
    eth-did = "did:eth:" eth-specific-identifier
```

File: did-eth-method-draft.md [hunk 2]
```diff
@@ -16,15 +16,51 @@
...
 
-### Syntax and Interpretation
+The namestring that shall identify this DID method is: `eth`
 
-```
-eth-did    = "did:eth:<chain_id>:" address
-chain_id   = chain_id according to [CAIP-2]
-address    = account_id according to [CAIP-10]
-```
+A DID that uses this method MUST begin with the following prefix: `did:eth`. Per the DID specification, this string
```
Comment (mirceanis) [inline did-eth-method-draft.md:23]:
```suggestion
A DID that uses this method MUST begin with the following prefix: `did:eth:`. Per the DID specification, this string
```

adding the colon to differentiate from did:ethr

File: did-eth-method-draft.md [hunk 3]
```diff
@@ -16,15 +16,51 @@
...
+
+## Method Specific Identifier
+
+The method specific identifier is represented as the HEX-encoded secp256k1 public key (in compressed form),
+or the corresponding HEX-encoded Ethereum address on the target network, prefixed with `0x`.
+
+    eth-did = "did:ethr:" eth-specific-identifier
+    eth-specific-identifier = [ ethr-network ":" ] ethereum-address / public-key-hex / ens-name
+    eth-network = "mainnet" / "goerli" / network-chain-id
+    network-chain-id = "0x" *HEXDIG
```
Comment (mirceanis) [inline did-eth-method-draft.md:34]: do we want to keep the hex encoding for chain IDs? (can be raised as an issue and discussed separately

File: did-eth-method-draft.md [hunk 4]
```diff
@@ -16,15 +16,51 @@
...
+A DID that uses this method MUST begin with the following prefix: `did:eth`. Per the DID specification, this string
+MUST be in lowercase. The remainder of the DID, after the prefix, is specified below.
+
+## Method Specific Identifier
+
+The method specific identifier is represented as the HEX-encoded secp256k1 public key (in compressed form),
+or the corresponding HEX-encoded Ethereum address on the target network, prefixed with `0x`.
+
+    eth-did = "did:ethr:" eth-specific-identifier
+    eth-specific-identifier = [ ethr-network ":" ] ethereum-address / public-key-hex / ens-name
```
Comment (mirceanis) [inline did-eth-method-draft.md:32]:
```suggestion
    eth-specific-identifier = [ eth-network ":" ] ethereum-address / public-key-hex / ens-name
```

**Review by mirceanis — CHANGES_REQUESTED** [review]
suggested some typo fixes and raised some questions that can be discussed with the workgroup

**Review by otto-mora — APPROVED** [review]

File: did-eth-method-draft.md [hunk 5]
```diff
@@ -16,15 +16,51 @@
...
+
+    did:eth:mainnet:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045
+    did:eth:0x1:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045
+    did:eth:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045
+    did:eth:vitalik.eth
+
+If the identifier is a `public-key-hex`:
+
+- it MUST be represented in compressed form (see https://en.bitcoin.it/wiki/Secp256k1)
+- the corresponding `blockchainAccountId` entry is also added to the default DID document, unless the `owner` property
```
Comment (Vid201) [inline did-eth-method-draft.md:55]: Should we rename property 'owner' to 'controller' so it's more aligned with DID Core specs?

File: did-eth-method-draft.md [hunk 6]
```diff
@@ -16,15 +16,51 @@
...
+  has been changed to a different address.
+- all Read, Update, and Delete operations MUST be made using the corresponding `blockchainAccountId` and MUST originate
```
Comment (Vid201) [inline did-eth-method-draft.md:57]: For signature verification, we should support EIP-1271 so it works with smart contract wallets (AA).

File: did-eth-method-draft.md (hunk 5 above)
Comment (mirceanis) [reply did-eth-method-draft.md:55]: That makes sense. We can start with the controller being the DID itself (or rather the address represented by the DID).

There's an overlap in responsibilities in the did-core spec: the controller also represents one or more other DIDs whose verification relationships should be considered for the current DID document as well.
I don't know if this should be explicitly stated in this spec too, or if this is a responsibility of the app performing the DID resolution to do it recursively

File: did-eth-method-draft.md (hunk 6 above)
Comment (nickreynolds) [reply did-eth-method-draft.md]: removed this language from the PR as it should be detailed in a different section. we can discuss it further in a future PR.

**Comment by nickreynolds** [comment]
@mirceanis @Reccetech updated this PR based on https://github.com/veramolabs/did-eth/issues/27 - if the 2 of you approve the new language we should probably merge?

**Pull request merged**